src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,178 @@
+    private static final int NEGATIVE_AGE = -15;
+    private static final int NON_VALID_EDGE_AGE = 17;
+    private static final String EMPTY_FIELD = "";
+    private static final int ZERO_AGE = 0;
+    private static final User FIRST_VALID_USER =
+            new User(FIFTEEN_CHARS_LOGIN, SIX_CHARS_PASSWORD, EIGHTEEN_AGES);
+    private static final User SECOND_VALID_USER =
+            new User(SIX_CHARS_LOGIN, SIX_CHARS_PASSWORD, EIGHTEEN_AGES);
+    private static final User THIRD_VALID_USER =
+            new User(SEVENTEEN_CHARS_LOGIN, NINE_CHARS_PASSWORD, NINETEEN_AGES);
+    private static final User FIRST_USER_FOR_REPEAT =
+            new User(SEVEN_CHARS_LOGIN, SYMBOL_SIX_CHARS_PASSWORD, NINETY_AGES);
+    private static final User SECOND_USER_FOR_REPEAT =
+            new User(SEVEN_CHARS_LOGIN, SYMBOL_SIX_CHARS_PASSWORD, NINETY_AGES);
+    private static final User NON_VALID_LOGIN_USER =
+            new User(THREE_CHARS_LOGIN, SIX_CHARS_PASSWORD, EIGHTEEN_AGES);
+    private static final User NON_VALID_AGE_USER =
+            new User(SIX_CHARS_LOGIN, SIX_CHARS_PASSWORD, NEGATIVE_AGE);
+    private static final User NON_VALID_PASSWORD_USER =
+            new User(SEVENTEEN_CHARS_LOGIN, THREE_CHARS_PASSWORD, NINETEEN_AGES);
+    private static final User NON_VALID_EDGE_LOGIN_USER =
+            new User(FIVE_CHARS_EDGE_LOGIN, EIGHT_CHARS_PASSWORD, TWENTY_FIVE_AGES);
+    private static final User NON_VALID_EDGE_AGE_USER =
+            new User(FIFTEEN_CHARS_LOGIN, EIGHT_CHARS_PASSWORD, NON_VALID_EDGE_AGE);
+    private static final User NON_VALID_EDGE_PASSWORD_USER =
+            new User(FIFTEEN_CHARS_LOGIN, FIVE_CHARS_EDGE_PASSWORD, TWENTY_FIVE_AGES);
+    private static final User NON_VALID_NULL_ALL_FIELDS_USER =
+            new User(null, null, null);
+    private static final User NON_VALID_NULL_PASSWORD_AND_AGE_USER =
+            new User(FIFTEEN_CHARS_LOGIN, null, null);
+    private static final User NON_VALID_NULL_AGE_USER =
+            new User(FIFTEEN_CHARS_LOGIN, EIGHT_CHARS_PASSWORD, null);
+    private static final User NON_VALID_EMPTY_FIELDS_USER =
+            new User(EMPTY_FIELD, EMPTY_FIELD, ZERO_AGE);
+    private static final User NON_VALID_NULL_USER = null;
+    private static final String NULL_FIELDS_EXCEPTION_MESSAGE =
+            "All fields should be filled.";
+    private static final String MIN_LOGIN_EXCEPTION_MESSAGE =
+            "The login should be at least 6 characters.";
+    private static final String MIN_AGE_EXCEPTION_MESSAGE =
+            "The age should be more than 18yo.";
+    private static final String MIN_PASSWORD_EXCEPTION_MESSAGE =
+            "The password should be at least 6 characters.";
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void register_validUser_ok() {
+        User registeredUser = registrationService.register(FIRST_VALID_USER);
+        assertNotNull(registeredUser);
```

Comment:
should be 2 tests, 1 - check user not null, 2 -check user have expected state or something